torch/csrc/jit/codegen/cuda/test/test_gpu_fused_reduction.cpp

```diff
@@ -1272,6 +1272,186 @@ TEST_F(NVFuserTest, FusionPersistentBNBackwardAllreduce_CUDA) {
       fe.kernel(), outputs, aten_inputs, {at_grad_input}, __LINE__, __FILE__);
 }
 
+TEST_F(NVFuserTest, FusionGroupedReductionReEntrant1_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeSymbolicTensor(2);
+  fusion.addInput(tv0);
+
+  auto tv1 = add(tv0, IrBuilder::create<Double>(1));
+  auto tv2 = sum(tv1, {0});
+
+  auto tv3 = add(tv0, IrBuilder::create<Double>(2));
+  auto tv4 = sum(tv3, {0});
+
+  auto tv5 = add(tv2, tv4);
+  fusion.addOutput(tv5);
+
+  groupReductions({tv2, tv4});
+
+  auto tv0_cache = tv0->cacheAfter();
+
+  const int vec = 2;
```

Comment:
Out of curiosity what does the heuristics select if it's not grouped? Can the heuristics/scheduler run after grouping?

Reply:
`scheduler_params` output:

```

===== Reduction Stats ========
total_reduction_numel: 99
total_iteration_numel: 999
vectorize_factor: 1
n_tensor_inputs: 1
max_input_dtype_size: 4
block(16, 16, 1)

===== Reduction Parameters ========

Red On Slow Dim

Iteration Domain: blockIdx.x / threadIdx.x / multiple reductions per block /
Inner Reduction Domain: cross block - threadIdx.y / unroll / factor 4
Launch Parameters: BlockDim.x = 16, BlockDim.y = 16, BlockDim.z = -1, GridDim.x = -1, GridDim.y = -1, GridDim.z = -1, Smem Size = 0

====================================


===== Reduction Stats ========
total_reduction_numel: 99
total_iteration_numel: 999
vectorize_factor: 1
n_tensor_inputs: 1
max_input_dtype_size: 4
block(16, 16, 1)

===== Reduction Parameters ========

Red On Slow Dim

Iteration Domain: blockIdx.x / threadIdx.x / multiple reductions per block /
Inner Reduction Domain: cross block - threadIdx.y / unroll / factor 4
Launch Parameters: BlockDim.x = 16, BlockDim.y = 16, BlockDim.z = -1, GridDim.x = -1, GridDim.y = -1, GridDim.z = -1, Smem Size = 0

====================================
```

Grouping before scheduling currently fails as expressions are changed from `ReducitonOp` to `GroupedReductionOp`. Just making it work would be likely just some mechanical changes.